src/rust/types/src/events.rs

```diff
@@ -41,22 +42,80 @@ pub enum CancelReason {
     ViolatedLimitPrice,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(rename_all = "snake_case")
+)]
+pub enum MakerEventType {
+    Cancel,
+    Change,
+    Evict,
+    Place,
+}
+
+impl TryFrom<u8> for MakerEventType {
```

Comment:
Note that `MakerEvent` and `TakerEvent` are deprecated (see market.move or the changelog: https://econia.dev/move/changelog)